task/dumper-api/src/lib.rs

```diff
@@ -7,17 +7,36 @@
 #![no_std]
 
 use derive_idol_err::IdolError;
+use gateway_messages::DumperError as GwDumperError;
+use hubpack::SerializedSize;
+use serde::{Deserialize, Serialize};
 use userlib::*;
 
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    FromPrimitive,
+    Eq,
+    PartialEq,
+    IdolError,
+    Serialize,
+    Deserialize,
+    SerializedSize,
+)]
+
 ///
 /// These constitute an interface between the RoT and the SP in that the
 /// error codes are interpreted by the dump agent and turned into dump agent
-/// failures.  Ultimately, these error conditions are merely informative,
-/// but it is important that a particular error condition not be falsely
-/// reported; error conditions can be added to or eliminated from this enum,
-/// but the values for extant conditions should not be altered.
+/// failures.  
+///
+/// These errors are also serialized and passed up to MGS, and as such they
+/// should not be re-ordered. New errors may be added to the end, but if they
+/// are they should be added to the `From<DumperError> for GwDumperError` below
```

Comment:
This sounds right to me. I think long-term we probably want to support taking dumps via MGS, but on the hubris side that would mean MGS would talk directly to `dump-agent` (using its UDP protocol) to do so.

Agreed that changing `DumperError` is a technically compatibility-breaking change in MGS but that making it now is fine, since we haven't deployed any SP builds that know about it (and there haven't been any subsequent changes yet).

Reply:
FYI, the reason I put `DumperError` in `gateway-messages` is because I was translating all `SprotError` variants to `gateway_messages::SpError`. However, this isn't strictly necessary. I suppose I can panic [here](https://github.com/oxidecomputer/hubris/pull/1279/files#diff-58d9279309315e9491ef32635decc0290285d1745708be04539ba3cd25a92d31R39) instead, which would prevent us from re-adding this error.

Reply:
Removed from gateway-messages in https://github.com/oxidecomputer/management-gateway-service/pull/85

I'll remove here in a follow up once that merges.